storage/sealer/sched.go

```diff
@@ -381,6 +396,51 @@ func (sh *Scheduler) Info(ctx context.Context) (interface{}, error) {
 	}
 }
 
+func (sh *Scheduler) removeRequest(rmrequest *rmRequest) {
+
+	if sh.SchedQueue.Len() < 0 {
+		rmrequest.rmresE <- xerrors.New("No requests in the scheduler")
+	}
+
+	queue := sh.SchedQueue
+	for i, r := range *queue {
+		if r.SchedId == rmrequest.id {
+			queue.Remove(i)
+			rmrequest.rmresC <- struct{}{}
+		}
+	}
+	rmrequest.rmresE <- xerrors.New("No request with provided details found")
+}
+
+func (sh *Scheduler) RemoveRequest(ctx context.Context, schedId uuid.UUID) error {
+	retE := make(chan error)
```

Comment:
This channel must be buffered to guarantee that responding doesn't block, if it's not this may happen:
* RemoveRequest is called
* The request is sent, and removeRequest has started
* The context in RemoveRequest is cancelled and the method quits
* removeRequest / runSched goroutine hangs on `rmrequest.res <- ..`

```suggestion
	retE := make(chan error, 1)
```